gtest/src/system.rs

```diff
@@ -91,30 +106,37 @@ impl LazyPagesStorage for PagesStorage {
 /// ```
 pub struct System(pub(crate) RefCell<ExtManager>);
 
-impl Default for System {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
 impl System {
     /// Prefix for lazy pages.
     pub(crate) const PAGE_STORAGE_PREFIX: [u8; 32] = *b"gtestgtestgtestgtestgtestgtest00";
 
     /// Create a new testing environment.
+    ///
+    /// # Panics
+    /// Only one instance in the current thread of the `System` is possible to
+    /// create. Instantiation of the other one leads to runtime panic.
+    #[allow(clippy::new_without_default)]
```

Comment:
I think default can be returned back. Trait implementation can be documented, too

Reply:
Well, I disagree here.
Creating more than one instance of the type leads to panic. When you have `Default` trait you can't fully control instantiation, because you have now more than one instantiation points: `new` and `default` functions.

Reply:
Also the trait `#[derive]` stuff can cause some problems if someone decides to use `System` as an owned by the other `Default` implementor type.